### PR TITLE
(5) Cookie Banner

### PR DESCRIPTION
**Pull request by pezholio**
This adds a Cookie Banner to the site that enables all cookie-setting JavaScript (in our case Google Analytics) only if the user accepts.

It assumes that all cookie-using JS has a `data-cookies` element set and a `type="text/plain”` attribute (that prevents the JS from running).

The code then checks for the presence of a `hasConsent` cookie and  shows a banner if it has not been set. When a user hits `accept` the cookie is set.

If the cookie is set, all JS with a `data-cookies` attribute is activated by copying the script, and setting a type of `text/javascript`, which allows it to load.

This approach is lovingly nicked from https://github.com/nhsuk/cookie-consent

There's also a bit of extra Rollup/Babel config that I've added to make sure the JS works in all browsers (including IE11)

### Cookie Banner

image

### Cookies Page

image

### CODE REVIEW

File: app/views/layouts/_cookie_banner.html.erb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div id="cookieBanner" class="hidden">
+    By continuing to visit this site, you accept the use of cookies by Google Analytics for statistical purposes.
```
Comment (CristinaRO) [inline app/views/layouts/_cookie_banner.html.erb]: Can we make it so we don't collect stats if the user continues to use the site without selecting "Accept"?

I'm not a fan of sites that take ignoring the button as implicit consent.

If this is already the case, can we rephrase the part that implies data is being collected just by "continuing to visit this site"?

Comment (pezholio) [reply app/views/layouts/_cookie_banner.html.erb]: Whoops. That was a bad copy/paste decision. I agree, that's a Very Bad approach, and not one that's reflected in the code.

**Comment by CristinaRO** [comment]
[Explore cookies banner / plausible](https://trello.com/c/qt2UwfiL/5-explore-cookies-banner-plausible)

**Comment by CristinaRO** [comment]
[Cookie page](https://trello.com/c/rMmUXwgj/21-cookie-page)

File: app/views/layouts/_cookie_banner.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<div id="cookieBanner" class="hidden">
+    <p>We use some essential cookies to make this service work.</p>
+    <p>We'd also like to use analytics cookies so we can understand how you use the service and make improvements.</p>    <a href="/cookies" id="cookies-eu-more">Read more</a>
+    <button id="rejectCookies">Reject</button>
```
Comment (CristinaRO) [inline app/views/layouts/_cookie_banner.html.erb]: What's the difference between "Reject" and "Reject analytics cookies" in practical terms? Are we able to not set `session[:uuid]` and `session[:just_rated_place_id]`, and do want to offer this as an option?

And what about the cookie that remembers they rejected cookies? (😬 )

Comment (pezholio) [reply app/views/layouts/_cookie_banner.html.erb]: Whoops - that first bit was a rebase gone bad. I'll need to add the `just_rated_place_id` to the strictly necessary cookies on the cookie page.

**Review by CristinaRO — APPROVED** [review]
Great work, thank you!

**Pull request merged**